src/domain/accounts/domain/src/entities/account.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, TimeZone, Utc};
+use lazy_static::lazy_static;
+use opendatafabric::{AccountID, AccountName};
+use serde::{Deserialize, Serialize};
+
+///////////////////////////////////////////////////////////////////////////////
+
+// TODO: have some length restrictions (0 < .. < limit)
+pub type AccountDisplayName = String;
+
+pub const DEFAULT_ACCOUNT_NAME_STR: &str = "kamu";
+pub const DEFAULT_AVATAR_URL: &str = "https://avatars.githubusercontent.com/u/50896974?s=200&v=4";
+pub const DEFAULT_EMAIL: &str = "kamu@example.com";
+
+lazy_static! {
+    pub static ref DEFAULT_ACCOUNT_NAME: AccountName =
+        AccountName::new_unchecked(DEFAULT_ACCOUNT_NAME_STR);
+    pub static ref DEFAULT_ACCOUNT_ID: AccountID =
+        AccountID::new_seeded_ed25519(DEFAULT_ACCOUNT_NAME_STR.as_bytes());
+    pub static ref DUMMY_REGISTRATION_TIME: DateTime<Utc> =
+        Utc.with_ymd_and_hms(2024, 4, 1, 12, 0, 0).unwrap();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Account {
+    pub id: AccountID,
+    pub account_name: AccountName,
+    pub email: Option<String>,
+    pub display_name: AccountDisplayName,
+    pub category: AccountCategory,
+    pub origin: AccountOrigin,
+    pub avatar_url: Option<String>,
+    pub registered_at: DateTime<Utc>,
+    pub is_admin: bool,
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum AccountCategory {
+    User,
+    Organization,
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum AccountOrigin {
+    Cli,
+    Github(AccountOriginGithub),
```

Comment:
Is "origin" different from "provider"? Do we need another concept?

Reply:
I like your suggestion to call this "provider".

Reply:
Renamed everywhere